chain/beacon/convert.go

```diff
@@ -2,14 +2,16 @@ package beacon
 
 import (
 	"github.com/drand/drand/chain"
+	"github.com/drand/drand/protobuf/common"
 	proto "github.com/drand/drand/protobuf/drand"
 )
 
-func beaconToProto(b *chain.Beacon) *proto.BeaconPacket {
+func beaconToProto(b *chain.Beacon, beaconID string) *proto.BeaconPacket {
 	return &proto.BeaconPacket{
 		PreviousSignature: b.PreviousSig,
 		Round:             b.Round,
 		Signature:         b.Signature,
+		Metadata:          &common.Metadata{BeaconID: beaconID},
```

Comment:
ahhh spicy catch - I was looking at a similar thing for the partial signature problems but it looked fine there

Reply:
It was actually already there in the code, but just done manually